Content.Shared/Contraband/ContrabandSystem.cs

```diff
@@ -68,17 +83,20 @@ private void OnExamined(Entity<ContrabandComponent> ent, ref ExaminedEvent args)
                 args.PushMarkup(Loc.GetString(severity.ExamineText));
             }
 
+
             // text based on ID card
             List<ProtoId<DepartmentPrototype>>? departments = null;
+            var jobId = "";
+
             if (_id.TryFindIdCard(args.Examiner, out var id))
             {
                 departments = id.Comp.JobDepartments;
+                jobId = Loc.GetString(id.Comp.JobIcon.Id).Replace("JobIcon", string.Empty);
```

Comment:
Getting the string from the job icon is an extremely hacky way to do this and won't work everywhere. If a job or jobicon is ever renamed this will break.

Reply:
I thought the exact same thing, but I'm not sure how else you can get someone's job rather than their role. Someone's job is effectively equivalent to their ID, but it doesn't appear as though there's any storage for their job within the ID component aside from the icon, as the JobTitle field is the one that's set manually and is even less reliable as a result. This method of getting job from ID is used elsewhere in the code, which is why I finally settled on using it, but if there's a cleaner way to get someone's job please let me know because I also hate this method and find it hacky.

Reply:
The reason no job prototype is stored inside the ID is custom job titles assigned by the ID card console. The best way to approach this is probably to compare the `LocalizedJobTitle` string in `IdCardComponent` to the `LocalizedName` string in `JobPrototype`. I think that should work in all languages and even if a job title is changed.

Reply:
I tried this prior, and now that I'm testing it again, I'm running into the same problem where LocalizedJobTitle is an empty string even when the icon knows we're a bartender 
![image](https://github.com/user-attachments/assets/419cbdfc-4d56-40ff-af86-9c5c2ee37474)
If necessary, I can implement a rework to the ID component to add a job field, or ensure that LocalizedJobTitle is always filled, I'm just wary of scope creep.

Just say the word, and I'll get working on that.

Reply:
LocalizedJobTitle might not be shown correctly there because it is a property and not a variable.
![grafik](https://github.com/user-attachments/assets/4383f542-642c-4d7d-90a4-bf1ecce8f9bc)

It might also be caused by the variable not being networked. Instead of using your IDE, have a look at the DataFields using VV on both client and server to check what is happening there.

Reply:
using VV, it seems like the JobTitle field is null for both the server and client

Server: 
![image](https://github.com/user-attachments/assets/aad8dacd-577c-4e22-b2eb-bed2cfc45a15)


Client: 
![image](https://github.com/user-attachments/assets/b55e9396-d1fc-4532-8f6b-60143951a297)

Reply:
Does examining the ID work?
Because that uses LocalizedJobTitle as well
![grafik](https://github.com/user-attachments/assets/c2b4bbae-27bb-4002-8854-169da7b28b3d)

Reply:
Examining the ID doesn't seem to trigger breakpoints in IdExaminableSystem.cs.

I'll try other uses of LocalizedJobTitle

Reply:
It works here, weirdly. Very strange. I'm wondering if I'm just getting the wrong ID component somehow. 
![image](https://github.com/user-attachments/assets/40175099-a944-48ed-a745-b0c7d4708074)

Reply:
Ah, but here's an issue. Localized Job Title is getting the title from the field for job name, meaning it'll break if HoP gives a custom job title. Really seeming like the only consistent method is either using the icon, or adding a new field @slarticodefast 
![image](https://github.com/user-attachments/assets/adf272b3-eb5f-4e5b-93ab-09333badb26c)

Reply:
Using the icon is worse, since the HOP might give someone a certain job name, but set the icon wrong. Also the icon prototype names might not correspond to the job name in case they get changed in the future.

I think it is ok to have the system not work with custom jobs (or typos like "Bartenderr") as the examine text warns about "carrying this around without a good reason", making it an IC issue. Players should be able to tell that for example a "Clown Lawyer" would be allowed to have lawyer restricted contraband with them.

Reply:
My thought was that the icons work because the primary enforcement of contraband classes are security, and security sees job icons with sec glasses, so whether or not you'll get in trouble for having a piece of contraband is likely most dependent on the icon that security sees. Additionally, with the icon setup, it's effectively an Enum, since at least one icon has to be selected, while custom job names can have no allowed jobs (a brig medic marked as medic will be allowed to carry medical contra, while a brig medic named "brig medic" in the name based system will see that all contraband is disallowed to them).

If you're set on the LocalizedJobTitle version though, I'll try and figure out what's wrong and implement it that way once I find the issue.

Reply:
Been working on this for a few hours and just can't seem to figure it out. I think the ID in someone's inventory must just be different from the ID within the ID Computer Console, but I can't seem to figure out where exactly the swap is happening and the new ID is being generated. Calling it a night, I'll try and get back to this another day.

Reply:
Well, I lied, because I just couldn't let it go. You were right about the issue being networking. If I set the JobTitle field whenever LocalizedJobTitle is set, I can see it properly. I want to use the LocalizedJobTittle like you indicated, but I'm not sure how to make it a networked field. When I copy over 
[DataField]
[AutoNetworkedField]
from JobTitle and apply it to LocalizedJobTitle, my build fails with a message about IoC being called from the wrong thread, or something similar. I feel like there's probably a simple step I'm missing, but I'm not noticing anything big based on the Networking beginner's guide.